sdks/universal-router-sdk/src/entities/actions/uniswap.ts

```diff
@@ -139,18 +174,17 @@ export class UniswapTrade implements Command {
     let minimumAmountOut: BigNumber = BigNumber.from(
       this.trade.minimumAmountOut(this.options.slippageTolerance).quotient.toString()
     )
-
     // The router custodies for 3 reasons: to unwrap, to take a fee, and/or to do a slippage check
     if (routerMustCustody) {
+      const pools = this.trade.swaps[0].route.pools
+      const pathOutputCurrency = getPathCurrency(this.trade.outputAmount.currency, pools[pools.length - 1])
+      const pathOutputCurrencyAddress = pathOutputCurrency.isNative ? ETH_ADDRESS : pathOutputCurrency.address
+
       // If there is a fee, that percentage is sent to the fee recipient
       // In the case where ETH is the output currency, the fee is taken in WETH (for gas reasons)
       if (!!this.options.fee) {
         const feeBips = encodeFeeBips(this.options.fee.fee)
-        planner.addCommand(CommandType.PAY_PORTION, [
-          this.trade.outputAmount.currency.wrapped.address,
-          this.options.fee.recipient,
-          feeBips,
-        ])
+        planner.addCommand(CommandType.PAY_PORTION, [pathOutputCurrencyAddress, this.options.fee.recipient, feeBips])
```

Comment:
this is amazing, we can support the native eth fee at ur-sdk level. looking at [fee-collector](https://github.com/Uniswap/fee-collector), it seems like ur-sdk is way behind 😓. @hensha256 @ConjunctiveNormalForm wondering if we have any plan to keep the ur-sdk in [fee-collector](https://github.com/Uniswap/fee-collector) up-to-date?

separately, this reminds me of the need of testing UL fees against v4 swaps, after deploying to production networks. according to [chains.ts](https://github.com/Uniswap/fee-collector/blob/main/lib/util/chains.ts), this is one area that we cannot test e2e on Sepolia v4 right now.

Reply:
fwiw I asked @hensha256 if we can support ETH fees and she said yes. But I'm not up to date on all the details